### DISCUSSION TITLE
Error happen when using with UpdateRange Methord

### DISCUSSION THREAD

**Discussion by Mike6x** [Finbuckle/Finbuckle.MultiTenant#651] [Q&A]
Hi All,
 I using the Fullstact Hero template here:
               https://github.com/fullstackhero/dotnet-webapi-boilerplate 
              ( Finbuckle 6.10.0, Ardalis.Specification 6.1.0).  

-  When I try to add a list items to DB with  AddRangeAsync(items, cancellationToken) => It done. 
- But when using UpdateRangeAsync => It alway error : multitenantexception: '1 modified entities with tenant id not set.'

in  the SaveChange Async()  

    public override async Task<int> SaveChangesAsync(CancellationToken cancellationToken = new CancellationToken())
    {
         .....
        int result = await base.SaveChangesAsync(cancellationToken);
        ....
       return result;
    }

Please help !

**Comment by AndrewTriesToCode** [comment]
hi I'm not familiar with the fullstack hero stuff, but in general this implies that your new entities are not getting their TenantId property set. If you don't have an explicit TenantId property then it's using an implicit shadow property internally. On your dbcontext you can set an inherited proper called `TenantNotSetMode` to `TenantNotSetMode.Overwrite` and it should plug in the current tenant at the time of the insert to keep things working as desire.

Give that a try and let me know.

**Comment by Mike6x** [comment]
Hi Andrew,

Thanks for your answer!

I have no problem with any other methords like  Add, AddASync, AddRange, AddRangAsync, Update, UpdateAsync ... but only UpdateRange/UpdateRange Asyn so I guest this issure not in settings !!!

Regards

**Reply by AndrewTriesToCode** [reply]
Do you have a GitHub repo that reproduces that problem so I can see it happen? Did the mode I mentioned above solve the issue for you or it solved it for all those you listed except UpdateRange(Async)?

**Reply by Mike6x** [reply]
Hi Andrew,
Here are API & Client:

https://github.com/Mike6x/fullstackhero/tree/UpdateFinbuckle_6_1_0
https://github.com/Mike6x/blazor-wasm-boilerplate

I am trying with the CountiesController to import data from excel ( using AddRangAsync). 
- > When import_data have Id key = null , the importing succesfull
- > When import_data have Id Key = Id key of entity in Database, then error

- Excel sample data file in the API repository ( Countries_Insert_sample.exls, Countries_update_sample.exls)

Pls help !

**Comment by Mike6x** [comment]
Hi Andrew,

I founded the Ardalis Nuget using for Repository Pattern is reason. So I have wolked around by using dbset :

public class Repository : IRepository
{
    private readonly ApplicationDbContext _dbContext;

    public DapperRepository(ApplicationDbContext dbContext) => _dbContext = dbContext;
......
    public async Task UpdateRangeAsync<T>(IEnumerable<T> entities, CancellationToken cancellationToken = default)
    where T : class, IAggregateRoot
    {
        _dbContext.TenantNotSetMode = TenantNotSetMode.Overwrite;
        _dbContext.Set<T>().UpdateRange(entities);
        await _dbContext.SaveChangesAsync(cancellationToken);
    }
}
